datafusion-examples/examples/flight_sql_server.rs

```diff
@@ -0,0 +1,618 @@
+        _request: Request<Streaming<FlightData>>,
+    ) -> Result<i64, Status> {
+        info!("do_put_prepared_statement_update");
+        // statements like "CREATE TABLE.." or "SET datafusion.nnn.." call this function
+        // and we are required to return some row count here
+        Ok(-1)
+    }
+
+    async fn do_action_create_prepared_statement(
+        &self,
+        query: ActionCreatePreparedStatementRequest,
+        request: Request<Action>,
+    ) -> Result<ActionCreatePreparedStatementResult, Status> {
+        let user_query = query.query.as_str();
+        info!("do_action_create_prepared_statement: {user_query}");
+
+        let ctx = self.get_ctx(&request)?;
+        let testdata = datafusion::test_util::parquet_test_data();
+
+        // register parquet file with the execution context
+        ctx.register_parquet(
+            "alltypes_plain",
+            &format!("{testdata}/alltypes_plain.parquet"),
+            ParquetReadOptions::default(),
+        )
+        .await
+        .map_err(|e| status!("Error registering table", e))?;
+
+        let plan = ctx
+            .sql(user_query)
+            .await
+            .and_then(|df| df.into_optimized_plan())
+            .map_err(|e| Status::internal(format!("Error building plan: {e}")))?;
+
+        // store a copy of the plan,  it will be used for execution
+        let plan_uuid = Uuid::new_v4().hyphenated().to_string();
+        self.statements.insert(plan_uuid.clone(), plan.clone());
+
+        let plan_schema = plan.schema();
+
+        let arrow_schema = (&**plan_schema).into();
+        let message = SchemaAsIpc::new(&arrow_schema, &IpcWriteOptions::default())
+            .try_into()
+            .map_err(|e| status!("Unable to serialize schema", e))?;
+        let IpcMessage(schema_bytes) = message;
+
+        let res = ActionCreatePreparedStatementResult {
+            prepared_statement_handle: plan_uuid.into(),
+            dataset_schema: schema_bytes,
+            parameter_schema: Default::default(),
+        };
+        Ok(res)
+    }
+
+    async fn do_action_close_prepared_statement(
+        &self,
+        handle: ActionClosePreparedStatementRequest,
+        _request: Request<Action>,
+    ) {
+        let handle = std::str::from_utf8(&handle.prepared_statement_handle);
+        if let Ok(handle) = handle {
+            info!("do_action_close_prepared_statement: removing plan and results for {handle}");
+            let _ = self.remove_plan(handle);
+            let _ = self.remove_result(handle);
+        }
+    }
+
+    async fn register_sql_info(&self, _id: i32, _result: &SqlInfo) {}
+}
+
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct FetchResults {
+    #[prost(string, tag = "1")]
+    pub handle: ::prost::alloc::string::String,
+}
+
+impl ProstMessageExt for FetchResults {
+    fn type_url() -> &'static str {
+        "type.googleapis.com/arrow.flight.protocol.sql.FetchResults"
```

Comment:
I recommend using something not in the arrow namespace

Maybe

```suggestion
        "type.googleapis.com/datatfusion.example.com.sql.FetchResults"
```

>

Reply:
fixed